src/nlp/load.py

```diff
@@ -530,3 +532,23 @@ def load_dataset(
     ds = builder_instance.as_dataset(split=split)
 
     return ds
+
+
+def concatenate_datasets(
+    dsets: List["Dataset"], info: Optional[Any] = None, split: Optional[Any] = None,
+):
+    """
+    Converts a list of :obj:``nlp.Dataset`` with the same schema into a single :obj:``nlp.Dataset``.
+
+    Args:
+        dsets (:obj:``List[nlp.Dataset]``): A list of Datasets to concatenate
+        features (:obj:``nlp.Features``, `optional`, defaults to :obj:``None``): If specified, the features types of the dataset
+        info (:obj:``nlp.DatasetInfo``, `optional`, defaults to :obj:``None``): If specified, the dataset info containing info like
+            description, citation, etc.
+        split (:obj:``nlp.NamedSplit``, `optional`, defaults to :obj:``None``): If specified, the name of the dataset split.
+    """
+    if not all([dset.schema == dsets[0].schema for dset in dsets]):
+        raise ValueError("Schema must match for all datasets")
+    table = pa.concat_tables([dset._data for dset in dsets])
+    data_files = list(itertools.chain.from_iterable([dset._data_files for dset in dsets]))
+    return Dataset(table, info=info, split=split, data_files=data_files)
```

Comment:
I think it would be nice to try keep some info by default instead of losing them by default if possible.
We try to care a lot about dataset metadata and citations in the library.

Maybe we could concatenate most of the dataset info, attributes by attributes?
E.g. the citations can be concatenated, the description probably as well.

Reply:
Seems like we want a `merge_dataset_infos()` method. I could take a crack, but you're probably better suited to tinker around with what format you like best.

Reply:
Done, let me know if that's what you had in mind.